src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Protocol/AppIntent.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+using MorganStanley.Fdc3;
+
+namespace MorganStanley.ComposeUI.Fdc3.DesktopAgent.Protocol;
+
+public class AppIntent : IAppIntent
+{
+    public IIntentMetadata Intent { get; set; }
+
+    public IEnumerable<IAppMetadata> Apps { get; set; }
```

Comment:
I'd rather use the implementation type in these properties with the interface property implemented explicitly, eg.

```cs
public IEnumerable<AppMetadata> Apps { get; set; }
IEnumerable<IAppMetadata> IAppIntent.Apps => Apps;
```

Then you can get rid of the explicit casts needed when converting to/from `IEnumerable<IAppMetadata>` etc.
This applies to the other types in this namespace as well.